app/Livewire/GlobalSearch.php

```
@@ -419,22 +599,728 @@ private function loadSearchableItems()
+            'name' => $s->name,
+            'description' => $s->description,
+        ])->toArray();
+        $this->loadingServers = false;
+
+        // Auto-select if only one server
+        if (count($this->availableServers) === 1) {
+            $this->selectServer($this->availableServers[0]['id']);
+        }
+    }
+
+    public function selectServer($serverId, $shouldProgress = true)
```

PHPMD (2.15.0)

The method selectServer has a boolean flag argument $shouldProgress, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)